auth-server/src/main/java/com/fiftyfive/keycloak/UsersApiService.java

```diff
@@ -0,0 +1,63 @@
+package com.fiftyfive.keycloak;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.keycloak.broker.provider.util.SimpleHttp;
+import org.keycloak.models.KeycloakSession;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class UsersApiService {
+    private KeycloakSession session;
+    private final String ID = "id";
+    private final String USERNAME = "username";
+    private final String FIRSTNAME = "firstName";
+    private final String LASTNAME = "lastName";
+    private final String EMAIL = "email";
+    private final String PASSWORD = "password";
+    private final String BASE_URI = "http://localhost:8080/users/";
+
+    public UsersApiService(KeycloakSession session) {
+        this.session = session;
+    }
+
+    User getUserDetails(String username) {
```

Comment:
Fetching details in the getAllUsers() will achieve the goal. I don't think there is need for this function. It's just increasing the heap memory. No need to call api in both function. Or is there any other use of this function ?

Reply:
Let say you update some details of a specific user in user base , and if you want to get updated it on keycloak too, you don't call the whole users , it will be time consuming as we can have large number of users, other thing is that getUserByUsername is used internally in keycloak in lot of this . If we don't have updated hashmap then we need to make a call for a particular user , In all these cases you can't make a call to getAllUsers() , it will consume a lot of memory and time.

Reply:
For that purpose aren't we using loadedusers ? That will not fullfill the requirement ? I guess that is used for caching purpose only and retrieving the loaded users only.

Reply:
loadedUsers is in memory , it will always don't have updated users. it resides on keycloak server,right.